### PR TITLE
docs: added operators and zfp section

### PR DESCRIPTION
**Pull request by vicentebolea**
cc: @chuckatkins

### CODE REVIEW

**Comment by vicentebolea** [comment]
@JasonRuonanWang I am adding the rst docs for the ZFP operator. I made a brief and dummy documentation for the Operators in general, I wonder if you can have a look and perhaps if needed add some missing info.

File: docs/user_guide/source/operators/operators.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+###################
+Supported Operators
+###################
+
+The Operator abstraction allows ADIOS2 to act upon the user application data,
+either from a ``adios2::Variable`` or a set of Variables in an ``adios2::IO``
+object.  Current supported operations are:
+
+1. Data compression/decompression, lossy and lossless.
+2. Callback functions (C++11 bindings only) supported by specific engines
```
Comment (JasonRuonanWang) [inline docs/user_guide/source/operators/operators.rst:10]: Callback operators are currently not used in any engines.

File: docs/user_guide/source/operators/ZFPCompressor.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+ZFP provides compressed-array classes that support high throughput read and
+write random access to individual array elements. ZFP also supports serial and
+parallel (OpenMP and CUDA) compression of whole arrays, e.g., for applications
+that read and write large data sets to and from disk.
+
+ADIOS2 provides a ``ZFPCompressor`` operator that lets you compress an
+decompress variables. Below there is an example of how to invoke
+``ZFPCompressor`` operator:
+
```
Comment (JasonRuonanWang) [inline docs/user_guide/source/operators/ZFPCompressor.rst:16]: The operator is called CompressZFP, also the name of the class

**Comment by JasonRuonanWang** [comment]
Looks all good to me, except for a few minor things which may or may not need to be changed.

**Comment by vicentebolea** [comment]
@JasonRuonanWang thanks for taking care of it, I was off for a few days and missed this, I Will follow up with another PR to address the previous points.

**Pull request merged**